### PR TITLE
pre-commit: test PR80983

### PR DESCRIPTION
**Pull request by dtcxzyw**
PR Link: https://github.com/llvm/llvm-project/pull/80983

### CODE REVIEW

File: bench/cpython/optimized/unicodeobject.ll [hunk 1]
```diff
@@ -7890,9 +7890,10 @@ if.then23:                                        ; preds = %if.then10
   ]
 
 land.lhs.true33:                                  ; preds = %if.then23
-  %spec.select.sroa.sel91.v = select i1 %cmp25, i64 5, i64 4
-  %spec.select.sroa.sel91 = getelementptr inbounds i8, ptr %buflower, i64 %spec.select.sroa.sel91.v
-  %10 = load i8, ptr %spec.select.sroa.sel91, align 1
+  %.sroa.gep110 = getelementptr inbounds i8, ptr %buflower, i64 5
+  %.sroa.gep110.val = load i8, ptr %.sroa.gep110, align 1
+  %incdec.ptr.val116 = load i8, ptr %incdec.ptr, align 1
+  %10 = select i1 %cmp25, i8 %.sroa.gep110.val, i8 %incdec.ptr.val116
```
Comment (dtcxzyw) [inline bench/cpython/optimized/unicodeobject.ll:7896]: Regression

File: bench/curl/optimized/libcurl_la-smtp.ll [hunk 2]
```diff
@@ -2008,8 +2008,9 @@ if.end:                                           ; preds = %lor.lhs.false17
 if.then34:                                        ; preds = %if.end
   %conv38 = select i1 %cmp32, i64 5, i64 3
   %7 = sub nuw nsw i64 6, %conv38
-  %8 = getelementptr i8, ptr %tmpline, i64 %conv38
-  call void @llvm.memset.p0.i64(ptr noundef nonnull align 1 dereferenceable(1) %8, i8 0, i64 %7, i1 false)
+  %conv38.sroa.sel.v = select i1 %cmp32, i64 5, i64 3
```
Comment (dtcxzyw) [inline bench/curl/optimized/libcurl_la-smtp.ll:2011]: It should be eliminated by EarlyCSE/GVN

File: bench/duckdb/optimized/ub_duckdb_constraints.cpp.ll [hunk 3]
```diff
@@ -657,8 +657,9 @@ entry:
   call void @llvm.memcpy.p0.p0.i64(ptr noundef nonnull align 8 dereferenceable(7) %1, ptr noundef nonnull align 1 dereferenceable(7) %cond, i64 %call.i.i, i1 false)
   %_M_string_length.i.i.i.i = getelementptr inbounds i8, ptr %base, i64 8
   store i64 %call.i.i, ptr %_M_string_length.i.i.i.i, align 8, !tbaa !17
-  %arrayidx.i.i.i = getelementptr inbounds i8, ptr %1, i64 %call.i.i
-  store i8 0, ptr %arrayidx.i.i.i, align 1, !tbaa !18
+  %call.i.i.sroa.sel.v.sroa.sel.v.sroa.sel.v.sroa.sel.v = select i1 %tobool.not, i64 23, i64 28
+  %call.i.i.sroa.sel.v.sroa.sel.v.sroa.sel.v.sroa.sel = getelementptr inbounds i8, ptr %base, i64 %call.i.i.sroa.sel.v.sroa.sel.v.sroa.sel.v.sroa.sel.v
```
Comment (dtcxzyw) [inline bench/duckdb/optimized/ub_duckdb_constraints.cpp.ll:661]: Regression?

File: bench/cpython/optimized/unicodeobject.ll (hunk 1 above)
Comment (nikic) [reply bench/cpython/optimized/unicodeobject.ll:7896]: Per https://github.com/llvm/llvm-project/blob/b0c6fc81fe132f20eed38bd836620dfcb5ac17e9/llvm/lib/Transforms/InstCombine/InstCombineLoadStoreAlloca.cpp#L1058 this is the preferred form right now. But maybe we should change it, now that other places have become better at handling load(select) (e.g. it has GVN support now).

**Pull request closed without merging**